nats/example_test.go

```diff
@@ -21,26 +20,37 @@ func ExamplePublisher() {
 }
 
 func ExampleSubscriber() {
-	ctx := context.TODO()
-	sub, err := bps.NewSubscriber(ctx, "nats://localhost:4222/?client_id=my_client&start_at=first")
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	subscriber, err := bps.NewSubscriber(ctx, "nats://localhost:4222/?client_id=my_client&start_at=first")
 	if err != nil {
 		panic(err.Error())
 	}
-	defer sub.Close()
-
-	// can be just context.WithCancel - `cancel` can be used to stop consuming:
-	ctx, cancel := context.WithTimeout(ctx, time.Second)
-	defer cancel()
+	defer subscriber.Close()
 
-	// will block till context is cancelled:
-	err = sub.Topic("topic").Subscribe(
-		ctx,
+	var subscription bps.Subscription
+	subscription, err = subscriber.Topic("topic").Subscribe(
 		bps.HandlerFunc(func(msg bps.SubMessage) {
 			_, _ = fmt.Printf("%s\n", string(msg.Data()))
+
+			// recipe to abort everything on irrecoverable handler errors:
+			_ = subscription.Close() // abort subscription, so current message won't be Ack-ed
+			cancel()                 // and cancel parent/program context to finally exit
 		}),
 		bps.StartAt(bps.PositionOldest),
 	)
 	if err != nil {
 		panic(err.Error())
 	}
+	defer subscription.Close()
+
+	<-ctx.Done() // block till subscription errors
+
+	// if interested in subscription error:
```

Comment:
I would just skip this (in all examples). an example shouldn't contain an overload of information but only the most necessary bits. you can add that comment to `Close` method of the Subscription interface